### PR TITLE
[ALS-5056] Create sumtype for QueryRequest

### PR DESCRIPTION
**Pull request by Luke-Sikina**
- Interface, implementations for general use, GIC

### CODE REVIEW

File: pic-sure-resources/pic-sure-resource-api/src/main/java/edu/harvard/dbmi/avillach/domain/GICQueryRequest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+				"        \"expectedResultType\": \"DATAFRAME\",\n" +
+				"        \"fields\": []\n" +
+				"    }\n" +
+				"}")
+
+/*
+ * QueryRequests for vanilla GIC PIC-SURE. Includes details about the source
+ * of the request, the requester, and any unifying identifiers for sharing
+ */
+public class GICQueryRequest implements QueryRequest {
```
Comment (JamesPeck) [inline pic-sure-resources/pic-sure-resource-api/src/main/java/edu/harvard/dbmi/avillach/domain/GICQueryRequest.java:47]: Can we name this something more generic? The implementation of a common area may one day be applied outside of GIC.

File: pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/security/JWTFilter.java [hunk 2]
```diff
@@ -204,7 +204,7 @@ private AuthUser callTokenIntroEndpoint(ContainerRequestContext requestContext,
 						edu.harvard.dbmi.avillach.data.entity.Resource resource = resourceRepo.getById(resourceUUID);
 						//logger.info("resource obj: " + resource + "    path: " + resource.getResourceRSPath());
 						if (resource != null && resource.getResourceRSPath() != null){
-							QueryRequest queryRequest = new QueryRequest();
+							GeneralQueryRequest queryRequest = new GeneralQueryRequest();
```
Comment (Luke-Sikina) [inline pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/security/JWTFilter.java:207]: Does the type need to be General?

File: pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureSearchService.java [hunk 3]
```diff
@@ -33,7 +34,7 @@ public class PicsureSearchService {
 	 * Executes a concept search against a target resource
 	 *
 	 * @param resourceId         - UUID of target resource
-	 * @param searchQueryRequest - {@link QueryRequest} containing resource specific credentials object
+	 * @param searchQueryRequest - {@link GeneralQueryRequest} containing resource specific credentials object
```
Comment (Luke-Sikina) [inline pic-sure-api-war/src/main/java/edu/harvard/dbmi/avillach/service/PicsureSearchService.java:37]: no

File: pic-sure-resources/pic-sure-ga4gh-dos/src/main/java/edu/harvard/hms/dbmi/avillach/GA4GHResourceRS.java [hunk 4]
```diff
@@ -2,13 +2,8 @@
 
 import java.io.IOException;
 import java.io.UnsupportedEncodingException;
-import java.lang.annotation.Annotation;
```
Comment (Luke-Sikina) [inline pic-sure-resources/pic-sure-ga4gh-dos/src/main/java/edu/harvard/hms/dbmi/avillach/GA4GHResourceRS.java:5]: exclude, no real changes

**Pull request merged**